site/app/libraries/database/DatabaseQueries.php

```diff
@@ -112,18 +112,32 @@ public function loadThreads($announcements){
         return $this->course_db->rows();
     }
 
+    public function createReply($user, $content, $thread_id, $anonymous, $type, $parent_post, $hasAttachment){
+        try {
+            $this->course_db->query("INSERT INTO posts (thread_id, parent_id, author_user_id, content, timestamp, anonymous, deleted, endorsed_by, resolved, type, has_attachment) VALUES (?, ?, ?, ?, current_timestamp, ?, ?, ?, ?, ?, ?)", array($thread_id, $parent_post, $user, $content, $anonymous, 0, NULL, 0, $type, true));
+            $this->course_db->query("DELETE FROM viewed_responses WHERE thread_id = ?", array($thread_id));
+            //retrieve generated thread_id
+            $this->course_db->query("SELECT MAX(id) as max_id from posts where thread_id=? and author_user_id=?", array($thread_id, $user));
+        } catch (DatabaseException $dbException){
+            if($this->course_db->inTransaction()){
+                $this->course_db->rollback();
+            }
+        }
+    }
+
     public function createPost($user, $content, $thread_id, $anonymous, $type, $first, $hasAttachment){
         $parent_post = -1;
         if(!$first){
             $this->course_db->query("SELECT id FROM posts where timestamp = (SELECT MAX(timestamp) from posts where thread_id = ? and deleted=false)", array($thread_id));
-            $parent_post = $this->course_db->rows()[0]["id"];
+            // $parent_post = $this->course_db->rows()[0]["id"];
+            $parent_post = 1; //The default posts on the thread all have parent_id of 1. 
```

Comment:
We should discuss this, I am not sure that assigning a default parent_id for replies to the master thread (especially one that can be a valid post id) is the best approach. It might make more sense to make the parent_id for posts that are a reply to the master thread be the post_id of the initial post. The other approach we can do is possibly look to add another column saying whether a post is a reply to the thread or to a post where it would enable a simple query to find the previous post that was a reply to the thread that could be assigned to the current post's parent_id.